core/server/api_container/server/startosis_engine/interpretation_time_value_store/interpretation_time_value_store.go

```diff
@@ -50,3 +56,32 @@ func (itvs *InterpretationTimeValueStore) RemoveService(name service.ServiceName
 	}
 	return nil
 }
+
+func (itvs *InterpretationTimeValueStore) PutServiceConfig(name service.ServiceName, serviceConfig *service.ServiceConfig) {
+	itvs.serviceConfigValues[name] = serviceConfig
+}
+
+func (itvs *InterpretationTimeValueStore) GetServiceConfig(name service.ServiceName) (*service.ServiceConfig, error) {
+	serviceConfig, ok := itvs.serviceConfigValues[name]
+	if !ok {
+		return nil, stacktrace.NewError("Did not find new service config for '%v' in interpretation time value store.", name)
+	}
+	return serviceConfig, nil
+}
+
+func (itvs *InterpretationTimeValueStore) SetServiceConfig(name service.ServiceName, serviceConfig *service.ServiceConfig) {
+	itvs.setServiceConfigValues[name] = serviceConfig
+}
+
+func (itvs *InterpretationTimeValueStore) ExistsNewServiceConfigForService(name service.ServiceName) bool {
+	_, doesConfigFromSetServiceInstructionExists := itvs.setServiceConfigValues[name]
+	return doesConfigFromSetServiceInstructionExists
+}
+
+func (itvs *InterpretationTimeValueStore) GetNewServiceConfig(name service.ServiceName) (*service.ServiceConfig, error) {
```

Comment:
This API doesn't account for multiple `set_service` instructions properly and its a bit complex, gonna fix and add tests